src/assets/js/copy-code.js

```diff
@@ -0,0 +1,52 @@
+    const clipboardSVG = '<svg width="24" height="24" viewBox="0 0 24 24" role="presentation" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M8 2H7.5V2.5V5.5C7.5 6.32843 8.17157 7 9 7H15C15.8284 7 16.5 6.32843 16.5 5.5V2.5V2H16H8ZM8.5 5.5V3H15.5V5.5C15.5 5.77614 15.2761 6 15 6H9C8.72386 6 8.5 5.77614 8.5 5.5ZM5 4H7V5H5L5 21.0027H19.0024V5H17V4H19.0024C19.5547 4 20.0024 4.44772 20.0024 5V21.0027C20.0024 21.555 19.5547 22.0027 19.0024 22.0027H5C4.44771 22.0027 4 21.555 4 21.0027V5C4 4.44771 4.44772 4 5 4Z" /></svg>';
+
+    const eventListener = () => {
+        if (document.readyState !== "complete") {
+            return
+        }
+
+        const copyHTML = copySVG + 'Copy';
+        const copyLabel = 'Copy to clipboard';
+
+        const codeBlocks = document.querySelectorAll('div.highlight');
+        codeBlocks.forEach((wrapperEl, i) => {
+            // find the <pre> element and assign it an ID for the button to reference
+            const codeEl = wrapperEl.querySelector('pre');
+            if (!codeEl) {
+                return;
+            }
+            if (!codeEl.id) {
+                codeEl.id = 'codeblock-' + i.toString();
+            }
+
+            // create a button and insert it inside the wrapper element
+            const copyBtn = document.createElement('button');
+            copyBtn.dataset.clipboardTarget = '#' + codeEl.id;
+            copyBtn.innerHTML = copyHTML;
+            copyBtn.classList.add('copy-to-clipboard');
+            copyBtn.setAttribute('aria-label', copyLabel)
+            wrapperEl.append(copyBtn);
+        });
+
+        const clipboard = new ClipboardJS('button.copy-to-clipboard');
+        clipboard.on('success', (e) => {
+            e.trigger.innerHTML = clipboardSVG + 'Copied!';
+            e.trigger.removeAttribute('aria-label');
+            e.clearSelection();
+
+            // restore the button state after a few seconds
+            setTimeout((target) => {
+                target.innerHTML = copyHTML;
+                target.setAttribute('aria-label', copyLabel);
```

Comment:
i'm curious about toggling the aria markup! is there a reason for that?

Reply:
I thought "Copy to clipboard" was more descriptive for AT (especially since, as you pointed out, there doesn't seem to be a great way to tie this to a specific element) than just "Copy" which is the visible text.

But once you have triggered it, the contents change (temporarily) to "Copied!" which should be sufficient context-wise on its own at that point, but if I left the `aria-label` as-is, it'd be mismatched, so I wanted to make sure the feedback wasn't obscured by the label.